pgcli/main.py

```diff
@@ -425,25 +439,37 @@ def adjust_less_opts(self):
 
         return less_opts
 
-    def refresh_completions(self):
-        self.completion_refresher.refresh(self.pgexecute, self.pgspecial,
-                                          self._on_completions_refreshed)
+    def refresh_completions(self, history=None, persist_priorities='all'):
+        callback = functools.partial(self._on_completions_refreshed,
+                                     persist_priorities=persist_priorities)
+        self.completion_refresher.refresh(
+            self.pgexecute, self.pgspecial, callback, history=history)
         return [(None, None, None,
                 'Auto-completion refresh started in the background.')]
 
-    def _on_completions_refreshed(self, new_completer):
-        self._swap_completer_objects(new_completer)
+    def _on_completions_refreshed(self, new_completer, persist_priorities):
+        self._swap_completer_objects(new_completer, persist_priorities)
 
         if self.cli:
             # After refreshing, redraw the CLI to clear the statusbar
             # "Refreshing completions..." indicator
             self.cli.request_redraw()
 
-    def _swap_completer_objects(self, new_completer):
+    def _swap_completer_objects(self, new_completer, persist_priorities):
         """Swap the completer object in cli with the newly created completer.
         """
         with self._completer_lock:
+            old_completer = self.completer
             self.completer = new_completer
+
+            if persist_priorities == 'all':
+                new_completer.prioritizer = old_completer.prioritizer
+            elif persist_priorities == 'keywords':
+                new_completer.prioritizer = old_completer.prioritizer
+                new_completer.prioritizer.clear_names()
+            elif persist_priorities == 'none':
+                pass
```

Comment:
Never mind, I'm able to follow it now. I still think a comment would be nice, since the function name says it's swapping completer objects and we actually do more than that inside the function.